test/savefiles_tests.jl

```diff
@@ -78,6 +78,24 @@ for ending ∈ ("bson", "jld2")
     rm(savename(simulation, ending))
     @test !isfile(savename(simulation, ending))
 
+    @test !isfile(savename(simulation, ending))
+    sim, path = produce_or_load(simulation; suffix = ending) do simulation
+        @test typeof(simulation.T) <: Real
+        a = rand(10); b = [rand(10) for _ in 1:10]
+        return @strdict a b simulation
+    end
+    @test isfile(savename(simulation, ending))
+    @test sim["simulation"].T == T
+    @test path == savename(simulation, ending)
+    sim, path = produce_or_load(simulation; suffix = ending) do simulation
```

Comment:
for safety here you should use the `produce_or_load(path, simulation; ...) do ...` syntax, so that both are checked.

Reply:
Should we do the same for the exisiting tests that don't use do-block as well then?

Reply:
I don't think that is necessary. Here it is simply not clear to me of `f(a, b) do x` what is `x`? is it the tuple `(a, b)` or is it only `a`? Or is it `b`?

Reply:
OK. I updated the test. Although it should not be an issue to people familar with do-blocks I suppose because they themselves have nothing to do with other arguments in the function call.

Reply:
You have written: `f(path, c) do c`, which confuses me, because `c` is the same symbol as the input argument. How does this work? What is `c`? The Julia docs provide the example:

```
open("outfile", "w") do io
    write(io, data)
end
```
here `io` is neither `"outfile"` nor `"w"`, so how in our case does what comes _after_ the `do` block coincide with the _input argument_ given to the function that initializes the `do` block?